docs/rules/require-await-function-call.md

```diff
@@ -0,0 +1,89 @@
+# require-await-function-call
+
+Some functions are asynchronous and we want to wait for their code to finish executing before continuing on. The modern `async` / `await` syntax can help us achieve this.
+
+## Rule Details
+
+This lint rule requires that specified functions (defaulting to the [async Ember test helpers](../../lib/utils/async-ember-test-helpers.js) like `click` and `visit`) be called with the `async` keyword. The benefits of this include:
+
+* Ensure code runs in the right (deterministic) order (potentially making tests more deterministic and reducing test flakiness)
+* Promote cleaner code by reducing unwieldy promise chain usage
+* Enforce a consistent way of calling/chaining asynchronous functions
+
+## Examples
+
+Examples of **incorrect** code for this rule:
+
+```js
+// Lint rule configuration: ['error', { functions: ['asyncFunc1', 'asyncFunc2'] }]
+function doSomethingInvalid() {
+  return asyncFunc1().then(() => {
+    return asyncFunc2();
+  });
+}
+```
+
+```js
+// Lint rule configuration: ['error', { functions: ['click'] }]
+test('clicking the button sends the action', function(assert) {
+  click('.my-button');
+  assert.ok(this.myAction.calledOnce);
+});
+```
+
+```js
+// Lint rule configuration: ['error', { functions: ['click'] }]
+test('clicking the button sends the action', function(assert) {
+  click('.my-button').then(() => {
+    assert.ok(this.myAction.calledOnce);
+  });
+});
+```
+
+Examples of **correct** code for this rule:
+
+```js
+// Lint rule configuration: ['error', { functions: ['asyncFunc1', 'asyncFunc2'] }]
+async function doSomethingValid() {
+  await asyncFunc1();
+  await asyncFunc2();
+}
+```
+
+```js
+// Lint rule configuration: ['error', { functions: ['click'] }]
+test('clicking the button sends the action', async function(assert) {
+  await click('.my-button');
+  assert.ok(this.myAction.calledOnce);
+});
+```
+
+## Options
+
+* OPTIONAL: `functions` is an array of the names of functions that must be called with `await`. Defaults to the [async Ember test helpers](../../lib/utils/async-ember-test-helpers.js) like `click` and `visit`.
+
+## Migration
+
+* [async-await-codemod](https://github.com/sgilroy/async-await-codemod) can help convert async function calls / promise chains to use `await`
+* [ember-test-helpers-codemod](https://github.com/simonihmig/ember-test-helpers-codemod) has transforms such as [click](https://github.com/simonihmig/ember-test-helpers-codemod/blob/master/transforms/acceptance/transforms/click.js) that can be modified to call `makeAwait()` and `dropAndThen()` on the function calls that you're trying to bring into compliance with this rule
+
+## When Not To Use It
```

Comment:
this should probably also mention testing loading states:

```js
click('.foo');
await waitFor('.loading-spinner');
assert.dom('.loading-spinner').isAwesome();
await settled();
```

in those cases you could still use this rule, but you would have to disable it for this specific case.